slate-video-library/src/Components/Context/HistorypageContext.js

```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import React, { createContext, useContext, useReducer } from "react";
+
+const historyContext = createContext({});
+export const useHistoryContext = () => useContext(historyContext);
+
+function HistorypageContext({ children }) {
+  function reducerFn(state, action) {
+    switch (action.type) {
+      case "ADD_TO_HISTORY":
+        return { ...state, addToHistory: action.payload };
+      case "GET_HISTORY":
+        return { ...state, getHistory: action.payload };
+      case "REMOVE_ALL_HISTORY":
+        return { ...state, removeCompleteHistory: action.payload };
+      case "REMOVE_ONEVIDEO_FROM_HISTORY":
+        return { ...state, removeVideoFromHistory: action.payload };
+      default:
+        return state;
+    }
+  }
+
+  const [state, setHistoryFn] = useReducer(reducerFn, {
+    addToHistory: [],
+    getHistory: [],
+    removeCompleteHistory: [],
+    removeVideoFromHistory: [],
+  });
+
+  const { getHistory } = state;
+  console.log(
+    "🚀 ~ file: HistorypageContext.js ~ line 29 ~ HistorypageContext ~ getHistory",
+    getHistory
+  );
+
+  const addToHistoryFn = async (videoData, setHistoryFn) => {
+    const response = await axios({
```

Comment:
you should use try catch method for error handling

Reply:
done